contracts/Rentals.sol

```diff
@@ -313,15 +274,30 @@ contract Rentals is NonceVerifiable, NativeMetaTransaction, IERC721Receiver {
     }
 
     /// @notice Standard function called by ERC721 contracts whenever a safe transfer occurs.
-    /// @dev The contract only allows safe transfers by itself made by the rent function.
-    /// @param _operator Caller of the safe transfer function.
+    /// Provides an alternative to acceptOffer by letting the asset holder send the asset to the contract
+    /// and accepting the offer at the same time.
+    /// @param _operator Caller of the safeTransfer function.
+    /// @param _from Address of the original holder of the token.
+    /// @param _data Bytes containing offer data.
     function onERC721Received(
         address _operator,
-        address, // _from,
-        uint256, // _tokenId,
-        bytes calldata // _data
-    ) external view override returns (bytes4) {
-        require(_operator == address(this), "Rentals#onERC721Received: ONLY_ACCEPT_TRANSFERS_FROM_THIS_CONTRACT");
+        address _from,
+        uint256 _tokenId,
+        bytes memory _data
+    ) external override returns (bytes4) {
+        if (_operator != address(this)) {
+            Offer memory offer = abi.decode(_data, (Offer));
+
+            // Check that the caller is the contract defined in the offer to ensure the function is being
+            // called through an ERC721.safeTransferFrom.
+            require(msg.sender == offer.contractAddress, "Rentals#onERC721Received: SENDER_CONTRACT_ADDRESS_MISMATCH");
+
+            // Prevents the holder from sending another token than the one defined in the offer.
+            require(offer.tokenId == _tokenId, "Rentals#onERC721Received: TOKEN_ID_MISMATCH");
+
+            _acceptOffer(offer, _from);
```

Comment:
I guess that this should be _operator instead of _from

Reply:
fixed

Reply:
_operator works fine when the holder of an asset sends it to the contract via `safeTransferFrom` because the operator is always the address of the caller of that function.

However, if the asset is being sent by another contract that was given allowance, _operator will be that contract instead of the holder.

If that was the case, that contract would be considered as the lessor, and would be the only one able to claim it back.

Using from makes it so that the holder of the token is set as lessor.

Reply:
Using `_operator` but added a comment in the contract to keep in mind a possible outcome